src/com/google/javascript/jscomp/deps/ModuleLoader.java

```diff
@@ -232,6 +248,14 @@ public ModulePath resolveJsModule(
 
     @Nullable
     private String resolveJsModuleFile(String moduleAddress) {
+      String[] fileExtensionsToSearch;
```

Comment:
How about passing the file extensions to search into this method instead?
I'd prefer to keep checking of ResolutionMode to only the one switch statement, even if it means creating some custom-named functions for particular code paths.
Such functions make it easier to know the context when you're reading.